pkg/download/downloader_test.go

```diff
@@ -140,3 +147,132 @@ func collectFiles(t *testing.T, scanPath string) []string {
 	}
 	return outFiles
 }
+
+func Test_extractContentType(t *testing.T) {
+	type args struct {
+		file string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "type zip",
+			args: args{
+				file: filepath.Join(testdataPath(), "test-with-directory.zip"),
+			},
+			want:    "application/zip",
+			wantErr: false,
+		},
+		{
+			name: "type tar.gz",
+			args: args{
+				file: filepath.Join(testdataPath(), "test-with-nesting-with-directory-entries.tar.gz"),
+			},
+			want:    "application/x-gzip",
+			wantErr: false,
+		},
+		{
+			name: "type elf-bin",
+			args: args{
+				file: filepath.Join(testdataPath(), "elf-file"),
+			},
+			want:    "application/octet-stream",
+			wantErr: false,
+		},
+		{
+			name: "type null",
```

Comment:
ditto. not sure why we're testing text/plain.

Reply:
This is for checking the behavior with 0-byte files, which could be an edge case.